File: OsmAnd/src/net/osmand/plus/myplaces/tracks/filters/TrackFilterList.java
package net.osmand.plus.myplaces.tracks.filters;

import androidx.annotation.Nullable;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.reflect.TypeToken;

import net.osmand.PlatformUtil;
import net.osmand.plus.track.data.SmartFolder;

import org.apache.commons.logging.Log;

import java.lang.reflect.Type;
import java.util.List;

public class TrackFilterList {
	private static final Log LOG = PlatformUtil.getLog(SmartFolderHelper.class);

	@Nullable
	public static List<SmartFolder> parseFilters(String str) {
		Gson gson = new GsonBuilder()
				.excludeFieldsWithoutExposeAnnotation()
				.registerTypeAdapter(BaseTrackFilter.class, new TrackFilterDeserializer())
				.create();

		Type token = new TypeToken<List<SmartFolder>>() {
		}.getType();
		List<SmartFolder> savedFilters = null;
		try {
			savedFilters = gson.fromJson(str, token);
		} catch (Throwable error) {
			error.printStackTrace();
			LOG.error(error.getMessage(), error);
		}
		return savedFilters;
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/filters/TrackFilterList.kt
package net.osmand.shared.filters

import kotlinx.serialization.json.Json
import net.osmand.shared.util.KAlgorithms
import net.osmand.shared.util.PlatformUtil


object TrackFilterList {
	fun parseFilters(str: String): List<SmartFolder>? {
		var savedFilters: List<SmartFolder>? = null
		if (!KAlgorithms.isEmpty(str)) {
			try {
				savedFilters = Json.decodeFromString<List<SmartFolder>?>(str)
			} catch (error: Throwable) {
				error.printStackTrace()
			}
		}
		return savedFilters
	}
}
